pkg/console/operator/status.go

```diff
@@ -0,0 +1,299 @@
+	updatedConfig, err := c.operatorConfigClient.UpdateStatus(operatorConfig)
+	if err != nil {
+		return nil, fmt.Errorf("status update error: %v \n", err)
+	}
+	return updatedConfig, nil
+}
+
+// Outputs the condition as a log message based on the detail of the condition in the form of:
+//   Status.Condition.<Condition>: <Bool>
+//   Status.Condition.<Condition>: <Bool> (<Reason>)
+//   Status.Condition.<Condition>: <Bool> (<Reason>) <Message>
+//   Status.Condition.<Condition>: <Bool> <Message>
+func logConditions(conditions []operatorsv1.OperatorCondition) {
+	logrus.Println("Operator.Status.Conditions")
+	for _, condition := range conditions {
+		buf := bytes.Buffer{}
+		buf.WriteString(fmt.Sprintf("Status.Condition.%s: %s", condition.Type, condition.Status))
+		hasMessage := condition.Message != ""
+		hasReason := condition.Reason != ""
+		if hasMessage && hasReason {
+			buf.WriteString(" |")
+			if hasReason {
+				buf.WriteString(fmt.Sprintf(" (%s)", condition.Reason))
+			}
+			if hasMessage {
+				buf.WriteString(fmt.Sprintf(" %s", condition.Message))
+			}
+		}
+		logrus.Println(buf.String())
+	}
+}
+
+// setStatusCondition
+// A generic helper for setting a status condition.
+// To use when another more specific status function is not sufficient.
+// examples:
+//  setStatusCondition(operatorConfig, Failing, True, "SyncLoopError", "Sync loop failed to complete successfully")
+func (c *consoleOperator) SetStatusCondition(operatorConfig *operatorsv1.Console, conditionType string, conditionStatus operatorsv1.ConditionStatus, conditionReason string, conditionMessage string) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               conditionType,
+		Status:             conditionStatus,
+		Reason:             conditionReason,
+		Message:            conditionMessage,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+// examples:
+//   conditionFailing(operatorConfig, "SyncLoopError", "Sync loop failed to complete successfully")
+func (c *consoleOperator) ConditionFailing(operatorConfig *operatorsv1.Console, conditionReason string, conditionMessage string) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeFailing,
+		Status:             operatorsv1.ConditionTrue,
+		Reason:             conditionReason,
+		Message:            conditionMessage,
+		LastTransitionTime: metav1.Now(),
+	})
+
+	return operatorConfig
+}
+
+func (c *consoleOperator) ConditionNotFailing(operatorConfig *operatorsv1.Console) *operatorsv1.Console {
+	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
+		Type:               operatorsv1.OperatorStatusTypeFailing,
+		Status:             operatorsv1.ConditionFalse,
+		LastTransitionTime: metav1.Now(),
```

Comment:
Ideally we'd add messages to all these, but OK as a follow on.

Reply:
`Failing:False` is the desired state, along with `Progressing:False` and `Available:True`.  I assumed if in the desired state no other information should be needed, but I'm open to adding info if we feel it is necessary or helpful.

Reply:
>  I assumed if in the desired state no other information should be needed, but I'm open to adding info if we feel it is necessary or helpful.

I think it would make things more clear if we add a message saying things are good. Particularly because you have to think through a double negative like failing: false. The messages are displayed on the cluster settings page in the UI.

(For a follow on)